mpas_analysis/test/test_mpas_xarray.py

```diff
@@ -0,0 +1,130 @@
+                                                             yearoffset=1850))
+        ds = mpas_xarray.subset_variables(ds, varList)
+        self.assertEqual(ds.data_vars.keys(), varList)
+        self.assertEqual(pd.Timestamp(ds.Time.values[0]),
+                         pd.Timestamp('1855-01-16 12:22:30'))
+
+        # next, test the same with the onlyvars argument
+        ds = xr.open_mfdataset(
+            fileName,
+            preprocess=lambda x: mpas_xarray.preprocess_mpas(x,
+                                                             timestr=timestr,
+                                                             onlyvars=varList,
+                                                             yearoffset=1850))
+        self.assertEqual(ds.data_vars.keys(), varList)
+
+    def test_iselvals(self):
+        fileName = str(self.datadir.join('example_jan.nc'))
+        timestr = 'time_avg_daysSinceStartOfSim'
+        varList = \
+            ['time_avg_avgValueWithinOceanLayerRegion_avgLayerTemperature',
+             'refBottomDepth']
+
+        iselvals = {'nVertLevels': slice(0, 3)}
+        ds = xr.open_mfdataset(
+            fileName,
+            preprocess=lambda x: mpas_xarray.preprocess_mpas(x,
+                                                             timestr=timestr,
+                                                             onlyvars=varList,
+                                                             iselvals=iselvals,
+                                                             yearoffset=1850))
+        self.assertEqual(ds.data_vars.keys(), varList)
+        self.assertEqual(ds[varList[0]].shape, (1, 7, 3))
+        self.assertEqual(ds['refBottomDepth'].shape, (3,))
+        self.assertApproxEqual(ds['refBottomDepth'][-1],
+                               4.882000207901)
+        date = pd.Timestamp(ds.Time.values[0])
+        # round to nearest second
+        date = pd.Timestamp(long(round(date.value, -9)))
+        self.assertEqual(date, pd.Timestamp('1855-01-13 12:24:14'))
+
+
+    def test_no_units(self):
+        fileName = str(self.datadir.join('example_no_units_jan.nc'))
+        timestr = 'time_avg_daysSinceStartOfSim'
+        varList = \
+            ['time_avg_avgValueWithinOceanLayerRegion_avgLayerTemperature',
+             'refBottomDepth']
+
+        ds = xr.open_mfdataset(
+            fileName,
+            preprocess=lambda x: mpas_xarray.preprocess_mpas(x,
+                                                             timestr=timestr,
+                                                             onlyvars=varList,
+                                                             yearoffset=1850))
+        self.assertEqual(ds.data_vars.keys(), varList)
+        date = pd.Timestamp(ds.Time.values[0])
+        # round to nearest second
+        date = pd.Timestamp(long(round(date.value, -9)))
+        self.assertEqual(date, pd.Timestamp('1855-01-13 12:24:14'))
+
+
+#    def test_selvals(self):
```

Comment:
@xylar, the issue is that the correct dimension to do the search on is `nVertLevels`, e.g., if you load the whole dateset without selection

```python
ds
<xarray.Dataset>
Dimensions:                                                      (Time: 1, nOceanRegionsTmp: 7, nVertLevels: 100)
Coordinates:
  * Time                                                         (Time) datetime64[ns] 1855-01-13T12:24:13.593599 ...
  * nOceanRegionsTmp                                             (nOceanRegionsTmp) int64 0 ...
  * nVertLevels                                                  (nVertLevels) int64 0 ...
Data variables:
    time_avg_avgValueWithinOceanLayerRegion_avgLayerTemperature  (Time, nOceanRegionsTmp, nVertLevels) float64 -0.6598 ...
    refBottomDepth                                               (nVertLevels) float64 1.51 ...
```
If I understand correctly, I think what we want is something more like

```python
ds.where(ds.refBottomDepth == 8.77999997138977, drop=True)
<xarray.Dataset>
Dimensions:                                                      (Time: 1, nOceanRegionsTmp: 7, nVertLevels: 1)
Coordinates:
  * Time                                                         (Time) datetime64[ns] 1855-01-13T12:24:13.593599 ...
  * nOceanRegionsTmp                                             (nOceanRegionsTmp) int64 0 ...
  * nVertLevels                                                  (nVertLevels) int64 4 ...
Data variables:
    time_avg_avgValueWithinOceanLayerRegion_avgLayerTemperature  (Time, nOceanRegionsTmp, nVertLevels) float64 -0.6544 ...
    refBottomDepth                                               (nVertLevels) float64 8.78 ...
```

This segements the dataset for all cases of `refBottomDepth` of `8.77999997138977`.  Is this the desired check here?  If so, I can fix this in the test.